python/tk_framework_desktopserver/shotgun/api_v1.py

```diff
@@ -26,11 +26,21 @@ class ShotgunAPI():
     Public API
     Callable methods from client. Every one of these methods can be called from the client.
     """
-    public_api = ["echo", "open", "executeToolkitCommand", "executeTankCommand",
-                  "pickFileOrDirectory", "pickFilesOrDirectories", "version",
-                  "getProjectActions"]
-
-    def __init__(self, host, process_manager):
+    PUBLIC_API_METHODS = [
+        "echo",
+        "open",
+        "executeToolkitCommand",
+        "executeTankCommand",
+        "pickFileOrDirectory",
+        "pickFilesOrDirectories",
+        "version",
+        "getProjectActions",
+    ]
+
+    # Nothing for v1. This is used in v2.
+    SYNCHRONOUS_METHODS = []
```

Comment:
I've not only because I've ignored the v1 API, but we can certainly give it a shot! It's a simple test.